hbase-balancer/src/main/java/org/apache/hadoop/hbase/master/balancer/BalancerClusterState.java

```diff
@@ -699,102 +684,57 @@ void regionMoved(int region, int oldServer, int newServer) {
     // update for servers
     int primary = regionIndexToPrimaryIndex[region];
     if (oldServer >= 0) {
-      primariesOfRegionsPerServer[oldServer] =
-        removeRegion(primariesOfRegionsPerServer[oldServer], primary);
+      removeElement(primariesOfRegionsPerServer.get(oldServer), primary, region);
```

Comment:
O(n)->O(1) for removing an element by its value. Please notice in the hash map the key is the value of the element of the old int[].